src/facdb.py

```diff
@@ -111,112 +119,127 @@ def geom_comparison(df, width=1000, height=600):
-        Use the slide bar and drop down to change the difference
-        threshold and select the attribute to review
+        qc_diff visualization
         """
-    )
+        thresh = st.sidebar.slider(
+            "difference threshold", min_value=0, max_value=300, value=5, step=1
+        )
+        level = st.sidebar.selectbox(
+            "select a classification level",
+            ["datasource", "factype", "facsubgrp", "facgroup", "facdomain"],
+            index=0,
+        )
+
+        st.sidebar.success(
+            """
+            Use the slide bar and drop down to change the difference
+            threshold and select the attribute to review
+            """
+        )
+        st.header(f"Change in number of records by {level}")
+        st.write(f"diff > {thresh}")
+
+        dff = qc_diff.groupby(level).sum()
+        dff = dff.loc[(dff["diff"] != 0) & (~dff["diff"].isna()), :]
+        if level == "factype":
+            st.warning(
+                "plot not available for this level,\
+                refer to the table below for more information"
+            )
+        else:
+            count_comparison(dff.loc[dff["diff"].abs() > thresh, :].sort_values("diff"))
 
-    st.header(f"Change in number of records by {level}")
-    st.write(f"diff > {thresh}")
+        st.header(f"Change in counts by {level}")
+        dff.insert(0, level, dff.index)
+        dff = dff.sort_values("diff")
+        plotly_table(dff)
 
-    dff = qc_diff.groupby(level).sum()
-    dff = dff.loc[(dff["diff"] != 0) & (~dff["diff"].isna()), :]
-    if level == "factype":
-        st.warning(
-            "plot not available for this level,\
-             refer to the table below for more information"
+        """
+        qc_mapped visualization
+        """
+        st.header(f"Change in percentage mapped records by {level}")
+        st.write(
+            """
+            Only instances where there is change in the percent
+            of mapped records and 100% of records are not mapped are reported
+        """
         )
-    else:
-        count_comparison(dff.loc[dff["diff"].abs() > thresh, :].sort_values("diff"))
-
-    st.header(f"Change in counts by {level}")
-    dff.insert(0, level, dff.index)
-    dff = dff.sort_values("diff")
-    plotly_table(dff)
-
-    st.header("New factypes")
-    st.write("Facility types that do not appear in the previous FacDB")
-    plotly_table(qc_diff.loc[qc_diff["count_old"] == 0, :])
-
-    st.header("Old factypes (retired)")
-    st.write(
-        "Facility types that do appear in the previous FacDB, \
-        but not in the latest version"
-    )
-    plotly_table(qc_diff.loc[qc_diff["count_new"] == 0, :])
+        dfff = qc_mapped.groupby(level).sum()
+        dfff.insert(0, level, dfff.index)
+        dfff["pctwogeom_old"] = dfff["wogeom_old"] / dfff["count_old"]
+        dfff["pctwogeom_new"] = dfff["wogeom_new"] / dfff["count_new"]
+        dfff["pctwogeom_old"] = dfff["pctwogeom_old"].round(2)
```

Comment:
When we settle on a more robust qa/qc process, we should have clear definitions towards the top of the facdb page for each metric we report